api/swagger.yaml

```diff
@@ -1577,6 +1577,20 @@ definitions:
           Settable: null
           Value: []
 
+  ObjectVersion:
+    description: |
+      The version number of the swarm related object such as node, service, etc. The version number is
+      needed to avoid conflicting writes. It is required that the client sends the version number along
+      with the modified specification for swarm related objects updates.
+      This approach ensures safe concurrency and determinism in that the change on the swarm related object
+      may not be applied if the version number has changed from the last read. As a result, two separate
+      update requests that happen at the same time will not unintentially overwrite each other.
```

Comment:
Perhaps be even more specific and say that if two update requests specify the same base version, only one of the requests can succeed.